### ISSUE TITLE
Failures in selfcal in the natural-weighting part of the difmap loop

### ISSUE THREAD

**Issue by nealjackson** [lmorabit/long_baseline_pipeline#68]
I've left some detailed comments on the selfcal channel of the slack workspace about this issue. There is a problem with the dif_script on some data I am processing, namely the part where the weighting becomes natural. Has anyone encountered the same thing? (Alternatively, if the dif_script as-is works for you, does it do so if you cut out the loop beginning with uvw 0,-1?)

**Comment by nealjackson** [comment]
Further investigations / pictures / comments now on the slack channel.

**Comment by nealjackson** [comment]
Further investigation: I think at least some of this is because the delay solutions are not good for some stations. Hypothesis: incoherent data is affecting the short baselines worse giving problems with naturally-weighted imaging. Now reinvestigating the delay solutions.

**Comment by lmorabit** [comment]
My suggested change is to go from:

```
    clean_selfcal_loop (fs,'2,0')
    clean_selfcal_loop (fs,'2,-1')
    clean_selfcal_loop (fs,'0,-1')

```
to 
```
    clean_selfcal_loop (fs,'2,0')
    clean_selfcal_loop (fs,'2,-1')
    if do_natural_weighting:
        clean_selfcal_loop (fs,'0,-1')
    else:
        clean_selfcal_loop (fs,'2,-1')
```

Is this the desired behaviour?

**Comment by nealjackson** [comment]
Yes, I think that is fine.

**Comment by lmorabit** [comment]
Fixed with commit ed8454e358b46c3f918ab7fa8fb264a7d1238472
